### PR TITLE
feat: logging audit — silent error paths and DEBUG diagnostics

### LINKED ISSUE #182: feat: general logging audit — silent error paths, consistent levels, startup banner
## Problem

Several code paths silently swallow errors (no logging at all), log levels are inconsistent across modules, and there is no startup configuration summary — making production debugging harder than necessary.

## Background

A logging audit of the codebase found 137 log statements across 10 files, but identified significant gaps:

**Silent error paths (no logging):**
- `collection._list_attachments()`: silently skips files on `ValueError`, `OSError`, stat errors
- `collection._validate_path()`: validation errors raised but not logged at point of detection
- `collection.reindex()` / `build_embeddings()`: per-document errors caught but some paths lack per-error logging with file path context

**Inconsistent log levels:**
- `config.py`: 19 DEBUG statements (good — configuration trace)
- `mcp_server.py`: 0 DEBUG statements (bad — no diagnostic detail)
- `fts_index.py`: only 7 statements for 600+ lines of core search logic
- `scanner.py`: mostly WARNING for parse errors, no INFO for successful operations

**Missing startup context:**
- No startup banner summarizing: log level, transport, auth mode, vault path, embedding provider, read-only vs read-write
- No indication of configuration source (env var values loaded)

Audit data: 48 DEBUG, 41 INFO, 37 WARNING, 11 ERROR, 0 CRITICAL across the codebase.

## Design Documentation

- [Design doc](docs/design.md), "Logging" section — specifies `logging.getLogger(__name__)` convention

## Acceptance Criteria

- [ ] 1. `collection._list_attachments()`: log WARNING when skipping a file due to `ValueError` or `OSError`, including the file path and exception message
- [ ] 2. `collection.reindex()` and `collection.build_embeddings()`: every per-document error logged at WARNING with the document path (verify no silent `except` blocks remain)
- [ ] 3. `fts_index.py`: add DEBUG logging for `search()` — query terms, result count, elapsed time
- [ ] 4. `scanner.py`: add INFO log for `parse_note()` success at DEBUG level (path, title, chunk count) — visible only with `LOG_LEVEL=DEBUG`
- [ ] 5. `vector_index.py`: add DEBUG logging for `search()` — query, k, result count
- [ ] 6. No silent `except` blocks remain in `collection.py` — every caught exception either re-raises or logs at WARNING or above with the file path context
- [ ] 7. `cli.py`: log at INFO level when CLI commands complete successfully (e.g., "Indexed N documents in M.Ms")

## Test Updates

- No new test files — existing tests must continue to pass
- Verify with `caplog` that previously-silent error paths now emit log messages (add targeted tests for the attachment listing skip case and the reindex per-document error case)
- Patch coverage must not regress

## Out of Scope

- `LOG_LEVEL` env var — tracked in separate issue
- Auth-specific logging — tracked in separate issue
- Structured JSON logging / log rotation — future enhancement
- Performance logging / metrics — separate concern

## Dependencies

- None — standalone (but pairs well with the LOG_LEVEL issue for end-to-end observability)

## Definition of Done

- [ ] All acceptance criteria checked off
- [ ] Tests pass, coverage not regressed
- [ ] architect-reviewer conformance table in PR body (all CONFORMANT)
- [ ] No silent `except` blocks in collection.py (verify by grep)

### PR DESCRIPTION
**Pull request by pvliesdonk**
## Summary

- Add WARNING logging to all previously-silent `except` blocks in `collection.py`
- Add DEBUG logging to `fts_index.search()`, `vector_index.search()`, and `scanner.parse_note()`
- Add INFO logging for CLI command completion (`index`, `search`, `reindex`)
- No silent `except` blocks remain in `collection.py`

Closes #182

**Stack:** This PR is stacked on #183 (auth observability). Merge #183 first, then retarget this to `main`.

## Conformance Table

| # | Requirement | Status |
|---|---|---|
| 1 | `_list_attachments`: WARNING on ValueError/OSError | CONFORMANT |
| 2 | `reindex()`/`build_embeddings()`: per-document errors at WARNING with path | CONFORMANT |
| 3 | `fts_index.search()`: DEBUG logging (query, filters, result count) | CONFORMANT |
| 4 | `scanner.parse_note()`: DEBUG log (path, title, chunk count) | CONFORMANT |
| 5 | `vector_index.search()`: DEBUG logging (query, limit, result count) | CONFORMANT |
| 6 | No silent `except` blocks in `collection.py` | CONFORMANT |
| 7 | CLI commands log at INFO on completion | CONFORMANT |

## Test plan

- [x] 2 new tests for previously-silent error paths (`caplog` verification)
- [x] 591 tests pass (full suite)
- [x] `grep -c 'except.*:' collection.py` — all blocks have logging

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test_collection.py [hunk 1]
```diff
@@ -3026,3 +3027,66 @@ def slow_callback(*args: object) -> None:
...
+                seen_csv = True
+            return original_stat(self_path, *a, **kw)
+
+        with caplog.at_level(logging.WARNING, logger="markdown_vault_mcp.collection"):
+            with patch.object(_Path, "stat", stat_that_fails):
+                results = col.list(include_attachments=True)
+        attachment_paths = [
+            r.path for r in results if isinstance(r, AttachmentInfo)
+        ]
+        assert "data.csv" not in attachment_paths
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply tests/test_collection.py]: Acknowledged — the `seen_csv` toggle is an implementation compromise. The approach you describe (splitting discovery from stat) isn't straightforward because `_list_attachments` does both rglob + stat in a single loop. The current test works correctly with the implementation as written. If the internal iteration pattern changes, we'd need to update the test anyway. Keeping as-is for now.

File: src/markdown_vault_mcp/collection.py [hunk 2]
```diff
@@ -1178,8 +1185,10 @@ def embeddings_status(self) -> dict:
...
                         else:
                             count = len(loaded_meta.get("rows", []))
-                    except (OSError, json.JSONDecodeError):
-                        pass
+                    except (OSError, json.JSONDecodeError) as exc:
+                        logger.warning(
+                            "embeddings_stats: could not read metadata — %s", exc
+                        )
 
         return {
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/markdown_vault_mcp/collection.py]: Fixed in cf61952. Changed `embeddings_stats` → `embeddings_status` to match the method name.

File: tests/test_collection.py [hunk 3]
```diff
@@ -3026,3 +3027,68 @@ def slow_callback(*args: object) -> None:
...
+            "path": "bad.md",
+            "title": "Bad",
+            "folder": "",
+            "frontmatter_json": "{broken-json",
+            "modified_at": 0.0,
+        }
+        with caplog.at_level(logging.WARNING, logger="markdown_vault_mcp.collection"):
+            result = _fts_row_to_note_info(row)
+        assert result.frontmatter == {}
+        assert any(
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply tests/test_collection.py]: Fixed in cf61952. Rewrote the test to call `Collection._get_frontmatter` directly via `patch.object(col._fts, "get_note", return_value=bad_row)` and assert `"_get_frontmatter: invalid JSON"` appears in caplog. The test now covers the actual new code path added in this PR.

**Comment by pvliesdonk** [comment]
Addressed all review comments in cf61952:

- **Fixed typo**: `embeddings_stats` → `embeddings_status` in log prefix (collection.py:1190)
- **Rewrote test**: `test_get_frontmatter_invalid_json_logs_warning` now tests `Collection._get_frontmatter` directly via `patch.object(col._fts, "get_note")` instead of the unrelated `_fts_row_to_note_info` helper. Asserts `"_get_frontmatter: invalid JSON"` in caplog.
- **Kept as-is**: `seen_csv` toggle in stat error test — works correctly with current implementation
- **Out of scope**: `embeddings_status` OSError/JSONDecodeError test coverage — rare edge case, simple code path

All 591 tests pass.
